gateway/ln-gateway/src/actor.rs

```diff
@@ -157,153 +235,124 @@ impl GatewayActor {
-                            Err(e) => {
-                                let fail = "Failed to parse payment hash";
-
-                                error!("{}: {:?}", fail, e);
-                                let _ = lnrpc_copy
-                                    .read()
-                                    .await
-                                    .complete_htlc(CompleteHtlcsRequest {
-                                        intercepted_htlc_id,
-                                        action: Some(Action::Cancel(Cancel {
-                                            reason: fail.to_string(),
-                                        })),
-                                    })
-                                    .await;
-                                continue;
+                        match action {
+                            Some(route_htlc_response::Action::SubscribeResponse(htlc)) => {
+                                Self::handle_intercepted_htlc(htlc, ln_sender.clone(), actor.clone()).await.expect("Error occurred while handling intercepted HTLC");
                             }
-                        };
-
-                        let amount_msat = Amount::from_msats(outgoing_amount_msat);
-
-                        let (outpoint, contract_id) = match actor
-                            .buy_preimage_from_federation(&hash, &amount_msat)
-                            .await
-                        {
-                            Ok((outpoint, contract_id)) => (outpoint, contract_id),
-                            Err(e) => {
-                                error!("Failed to buy preimage: {:?}", e);
-                                // Note: this specific complete htlc requires no further action.
-                                // If we fail to send the complete htlc message, or get an error
-                                // result, lightning node will still
-                                // cancel HTCL after expiry period lapses.
-                                // Result can be safely ignored.
-                                // TODO: make sure this succeeded?
-                                let _ = lnrpc_copy
-                                    .read()
-                                    .await
-                                    .complete_htlc(CompleteHtlcsRequest {
-                                        intercepted_htlc_id,
-                                        action: Some(Action::Cancel(Cancel {
-                                            reason: e.to_string(),
-                                        })),
-                                    })
-                                    .await;
-                                continue;
+                            Some(route_htlc_response::Action::CompleteResponse(_complete_response)) => {
+                                // TODO: Might need to add some error handling here
+                                info!("Successfully handled HTLC");
```

Comment:
Might be good to include some information about the HTLC that was handled

Reply:
Gonna leave this one as a todo since we don't actually know much about this HTLC (`CompleteResponse` doesn't contain any data currently).